pkg/apimachinery/patch/patch.go

```diff
@@ -38,6 +38,93 @@ const (
 	PatchRemoveOp  = "remove"
 )
 
+func (p *PatchOperation) MarshalJSON() ([]byte, error) {
+	switch p.Op {
+	// The 'remove' operation is the only patching operation without a value
+	// and it needs to be parsed differently.
+	case PatchRemoveOp:
+		return json.Marshal(&struct {
+			Op   string `json:"op"`
+			Path string `json:"path"`
+		}{
+			Op:   p.Op,
+			Path: p.Path,
+		})
+	case PatchTestOp, PatchReplaceOp, PatchAddOp:
+		return json.Marshal(&struct {
+			Op    string      `json:"op"`
+			Path  string      `json:"path"`
+			Value interface{} `json:"value"`
+		}{
+			Op:    p.Op,
+			Path:  p.Path,
+			Value: p.Value,
+		})
+	default:
+		return nil, fmt.Errorf("operation %s not recognized", p.Op)
+	}
+}
+
+type PatchSet struct {
+	patches []PatchOperation
+}
+
+type PatchOption func(patches *PatchSet)
+
+func New(opts ...PatchOption) *PatchSet {
+	p := &PatchSet{}
+	for _, f := range opts {
+		f(p)
+	}
+	return p
+}
+
+func (p *PatchSet) AddOption(ops ...PatchOption) {
```

Comment:
super nit: `ops` vs `opts`

Also not sure, makes sense to use this method in `New` above?

Reply:
Done